src/exporters/USDZExporter.ts

```diff
@@ -0,0 +1,457 @@
+      array,
+      8,
+    )}, ${this.buildMatrixRow(array, 12)} )`
+  }
+
+  private buildMatrixRow(array: number[], offset: number): string {
+    return `(${array[offset + 0]}, ${array[offset + 1]}, ${array[offset + 2]}, ${array[offset + 3]})`
+  }
+
+  // Mesh
+  private buildMeshObject(geometry: BufferGeometry): string {
+    const mesh = this.buildMesh(geometry)
+    return `
+def "Geometry"
+{
+  ${mesh}
+}
+`
+  }
+
+  private buildMesh(geometry: BufferGeometry): string {
+    const name = 'Geometry'
+    const attributes = geometry.attributes
+    const count = attributes.position.count
+
+    return `
+    def Mesh "${name}"
+    {
+        int[] faceVertexCounts = [${this.buildMeshVertexCount(geometry)}]
+        int[] faceVertexIndices = [${this.buildMeshVertexIndices(geometry)}]
+        normal3f[] normals = [${this.buildVector3Array(attributes.normal, count)}] (
+            interpolation = "vertex"
+        )
+        point3f[] points = [${this.buildVector3Array(attributes.position, count)}]
+        float2[] primvars:st = [${this.buildVector2Array(attributes.uv, count)}] (
+            interpolation = "vertex"
+        )
+        uniform token subdivisionScheme = "none"
+    }
+`
+  }
+
+  private buildMeshVertexCount(geometry: BufferGeometry): string {
+    const count = geometry.index !== null ? geometry.index.array.length : geometry.attributes.position.count
+
+    return Array(count / 3)
+      .fill(3)
+      .join(', ')
+  }
+
+  private buildMeshVertexIndices(geometry: BufferGeometry): string {
+    if (geometry.index !== null) {
+      // @ts-expect-error
+      return geometry.index.array.join(', ')
+    }
+
+    const array = []
+    const length = geometry.attributes.position.count
+
+    for (let i = 0; i < length; i++) {
+      array.push(i)
+    }
+
+    return array.join(', ')
+  }
+
+  private buildVector3Array(attribute: BufferGeometry['attributes'][number], count: number): string {
+    if (attribute === undefined) {
+      console.warn('USDZExporter: Normals missing.')
+      return Array(count).fill('(0, 0, 0)').join(', ')
+    }
+
+    const array = []
+    const data = attribute.array
+
+    for (let i = 0; i < data.length; i += 3) {
+      array.push(
+        `(${data[i + 0].toPrecision(this.PRECISION)}, ${data[i + 1].toPrecision(this.PRECISION)}, ${data[
+          i + 2
+        ].toPrecision(this.PRECISION)})`,
+      )
+    }
+
+    return array.join(', ')
+  }
+
+  private buildVector2Array(attribute: BufferGeometry['attributes'][number], count: number): string {
+    if (attribute === undefined) {
+      console.warn('USDZExporter: UVs missing.')
+      return Array(count).fill('(0, 0)').join(', ')
+    }
+
+    const array = []
+    const data = attribute.array
+
+    for (let i = 0; i < data.length; i += 2) {
+      // @ts-expect-error
+      array.push(`(${data[i + 0].toPrecision(this.PRECISION)}, ${1 - data[i + 1].toPrecision(this.PRECISION)})`)
```

Comment:
I wonder if it can work well...🤔 For now, I added `ts-expect-error`

Reply:
see above.